### ISSUE TITLE
[3.3 Beta] Dynamic data doesn't work properly for CPTs (multiple issues)

### ISSUE THREAD

**Issue by sunnyt7** [soflyy/oxygen-bugs-and-features#1059]
1. Create a new post type, and add a couple posts
2. Add a repeater to any page, and set the query to "custom" and select the post type you just created. (using default, custom, and manual queries all do not work)
3. Add a link wrapper inside the repeater div. For the URL, use Dynamic Data > Permalink. It will result in [oxygen data='permalink'].
4. Hit save and preview on front-end. You'll see it working.
5. Refresh the Oxygen editor. The URL on the link wrapper will not be set anymore. And if you hit save, that will obviously mean it doesn't link properly in the front-end anymore.

If I revert back to 3.2.1, the URL saves properly. It also seems to work if I'm just using regular posts.

Here is the sandbox link: https://oxygen-qa3sfj7qz558k.oxygen-demo.qsandbox.me/repeater-links/

Admin: support
Pass: support

**Comment by sunnyt7** [comment]
Looks like this actually isn't just limited to Repeaters. A regular link wrapper on the page that has a URL using dynamic data for a CPT gets cleared on refresh.

**Comment by sunnyt7** [comment]
@KittenCodes Sadly discovered today that this also actually extends past URLs 😢 

I have this function in Code Snippets:

```
function st_get_podcast_taxonomy_custom_field($field) {
	$terms = wp_get_post_terms( get_the_ID(), 'podcast_category', array( "fields" => "all" ) );
	return get_field($field, 'podcast_category_'.$terms[0]->term_id);
}
```
And then inside a CPT single post:

Text > Dynamic Data > PHP function, which outputted:

[oxygen data='phpfunction' function='st_get_podcast_taxonomy_custom_field' arguments='itunes_link']

This does not work in 3.3 beta. It results in a critical error, saying 0 arguments were passed.

Works fine in 3.2.1.

**Comment by KarzanTT** [comment]
+1

**Comment by KittenCodes** [comment]
Fixed in v3.3.
